data-node/src/main/java/org/graylog/datanode/bootstrap/preflight/OpensearchConfigSync.java

```diff
@@ -0,0 +1,111 @@
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Collections;
+
+public class OpensearchConfigSync implements PreflightCheck {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpensearchConfigSync.class);
+
+    private final Configuration configuration;
+
+    @Inject
+    public OpensearchConfigSync(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public void runCheck() throws PreflightCheckException {
+        final Path localOpensearchConfigDir = Path.of(configuration.getOpensearchConfigLocation()).resolve("opensearch");
+        LOG.info("Directory used for Opensearch configuration is {}", localOpensearchConfigDir.toAbsolutePath());
+
+        try {
+            Files.createDirectories(localOpensearchConfigDir);
+            synchronizeConfig(Path.of("opensearch", "config"), localOpensearchConfigDir);
+        } catch (IOException | URISyntaxException e) {
+            throw new RuntimeException("Failed to prepare opensearch config directory", e);
+
+        }
+    }
+
+    public void synchronizeConfig(Path configRelativePath, final Path target) throws URISyntaxException, IOException {
+        final URI uriToConfig = OpensearchConfigSync.class.getResource("/" + configRelativePath.toString()).toURI();
+        if ("jar".equals(uriToConfig.getScheme())) {
+            copyFromJar(configRelativePath, target, uriToConfig);
+        } else {
+            copyFromLocalFs(configRelativePath, target);
+        }
+    }
+
+    private static void copyFromJar(Path configRelativePath, Path target, URI uri) throws IOException {
+        try (
+                final FileSystem fs = FileSystems.newFileSystem(uri, Collections.emptyMap());
+        ) {
+            // Get hold of the path to the top level directory of the JAR file
+            final Path resourcesRoot = fs.getPath("/");
+            final Path source = resourcesRoot.resolve(configRelativePath.toString()); // caution, the toString is needed here to resolve properly!
+            copyRecursively(source, target);
+        }
+    }
+
+    private static void copyFromLocalFs(Path configRelativePath, Path target) throws URISyntaxException, IOException {
+        final Path resourcesRoot = Paths.get(OpensearchConfigSync.class.getResource("/").toURI());
+        final Path source = resourcesRoot.resolve(configRelativePath);
+        copyRecursively(source, target);
+    }
+
+    private static void copyRecursively(Path source, Path target) throws IOException {
```

Comment:
If we don't implement a full sync (including removals) right now, we should add a TODO here and create an issue in GitHub.